frontend/src/components/pages/CampCreation/CampDetails/index.tsx

```diff
@@ -280,6 +281,7 @@ const CampCreationDetails = ({
                 height="52px"
                 maxLength={2}
                 defaultValue={ageLower}
+                key={ageLower ? "loaded-age-lower" : "loading-age-lower"}
```

Comment:
I don't like this solution, but essentially defaultValue is only set on the first render (where the default values like ageLower = 0 is used). So afterwards, when the getCampById call returns us the current progress of a draft camp, the defaultValue doesn't update even though the underlying state updates. To solve this, we can change the key of the component whenever the data finishes loading. One of the issues with this is that the component looses focus every time we go from a truthy to falsy state and vice versa.

reference: https://stackoverflow.com/questions/30146105/react-input-defaultvalue-doesnt-update-with-state

Reply:
To be fair, this was endorsed by the React docs - https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key

I forced reload on a parent element, does this (https://github.com/uwblueprint/focus-on-nature/commit/786d56c0eacb2b7f9d3d2435a16657192f9618ed) look any better? Feel free to `git cherry-pick` that if it helps

Reply:
ahhh yeah this works soo much better. thanks!